app/change_sets/change_set.rb

```diff
@@ -76,4 +76,48 @@ def fix_multivalued_keys
     end
     @_changes = Disposable::Twin::Changed::Changes.new
   end
+
+  # Defines the default populator for a nested single-valued changeset property
+  def populate_nested_property(fragment:, as:, **)
+    property_klass = model.class.schema[as.to_sym]
+    if fragment.values.select(&:present?).blank?
+      send(:"#{as}=", nil)
+      return skip!
+    end
+
+    send(:"#{as}=", property_klass.new(fragment))
+  end
+
+  # Iterate through nested properties on a change set and call the defined
+  # populator method. The field values of the nested resource are passed to the
+  # populator as the fragment.
+  def populate_nested_properties
+    # Applying the twin filter to schema finds all nested properties
+    schema.each(twin: true) do |property|
+      name = property[:name]
+      populator = property[:populator]
+      # Try calling fields method on the property value
+      fields = send(name).try(:fields)
+      next if fields.nil? || fields.values.select(&:present?).blank?
+      send(populator, fragment: fields, as: name)
+    end
+  end
+
+  # Override prepopulate method to correctly populate nested properties.
+  def prepopulate!(_args = {})
+    schema.each(twin: true) do |property|
+      property_name = property[:name]
+      property_klass = model.class.schema[property_name.to_sym]
+      send(:"#{property_name}=", property_klass.new) unless send(property_name)
+    end
+
+    super
+  end
+
+  # Trigger population of nested properties when syncing changeset.
+  # Overrides {Disposable::Twin#sync}
+  def sync
```

Comment:
Why is this necessary if you're adding the populator to the necessary properties in the ChangeSet?

Reply:
The populator is not run when syncing the resource outside of a controller action. Instantiating a resource with the nested property in FactoryBot is an example. I'm sure why exactly, but I guess this is something buried within reform. Without the populator, when you run `sync` on the changeset, an exception is raised because reform (?) has mapped that nested property to a changeset.

Reply:
@tpendragon I created an branch with an example that raises the same exception with the existing letter change set. 

https://github.com/pulibrary/figgy/compare/nested-error-example

Reply:
Does that still error if for sender you instead set it as just the properties and not the object?